ui/components/multichain/network-manager/hooks/useNetworkChangeHandlers.ts

```diff
@@ -121,6 +120,28 @@ export const useNetworkChangeHandlers = () => {
         );
       }
 
+      const isDeselecting = Object.keys(enabledNetworksByNamespace).some(
+        (key) => key === hexChainId,
+      );
+
+      if (isGlobalNetworkSelectorRemoved && isDeselecting) {
+        const firstEnabledNetwork = enabledNetworkKeys[0];
+
+        const firstEnabledNetworkConfig = firstEnabledNetwork
+          ? evmNetworks[firstEnabledNetwork as keyof typeof evmNetworks]
+          : null;
+        const firstEnabledNetworkClientId =
+          firstEnabledNetworkConfig?.rpcEndpoints?.[
+            firstEnabledNetworkConfig.defaultRpcEndpointIndex
+          ]?.networkClientId;
```

Comment:
Couldn't this be undefined?

Reply:
Type-wise, this looks like it could be `undefined`, but you safely add a default below (`firstEnabledNetworkClientId || finalNetworkClientId`).

Will test tomorrow.